src/test/java/org/gitlab4j/api/TestStreams.java

```diff
@@ -85,4 +85,35 @@ public void testParallelStream() throws Exception {
             assertTrue(compareJson(sortedUsers.get(i), users.get(i)));
         }
     }
+
+    @Test
+    public void testLazyStream() throws Exception {
+
+        // Arrange
+        Pager<User> pager = new UserApi(gitLabApi).getUsers(10);
+        Stream<User> stream = pager.lazyStream();
+
+        // Assert
+        assertNotNull(stream);
+        List<String> usernames = stream.map(User::getUsername).collect(toList());
+        assertNotNull(usernames);
+
+        assertEquals(usernames.size(), sortedUsers.size());
+        for (int i = 0; i < sortedUsers.size(); i++) {
+            assertTrue(usernames.contains(sortedUsers.get(i).getUsername()));
+        }
+    }
+
+    @Test
+    public void testStreamLimit() throws Exception {
+
+        // Arrange
+        Stream<User> stream = new UserApi(gitLabApi).getUsersStream();
+
```

Comment:
This test has sense only for lazy streams, for normal stream it is guaranteed by Java API that this always pass, so an additional test is not needed. But for the lazy stream, it will be nice to add such test, with pagination lesser than a limit, because we need to be sure, that our splitter can gather data from gitlab.

Reply:
Good point, updated.